src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClientManager.cs

```diff
@@ -21,31 +23,79 @@ namespace Microsoft.Extensions.Configuration.AzureAppConfiguration
-            {
-                var endpoint = new Uri(ConnectionStringParser.Parse(connectionString, ConnectionStringParser.EndpointSection));
-                return new ConfigurationClientWrapper(endpoint, new ConfigurationClient(connectionString, clientOptions));
-            }).ToList();
-        }
-
-        public ConfigurationClientManager(IEnumerable<Uri> endpoints, TokenCredential credential, ConfigurationClientOptions clientOptions)
-        {
-            if (endpoints == null || !endpoints.Any())
-            {
-                throw new ArgumentNullException(nameof(endpoints));
-            }
-
-            _clients = endpoints.Select(endpoint => new ConfigurationClientWrapper(endpoint, new ConfigurationClient(endpoint, credential, clientOptions))).ToList();
+    {
+        private IList<ConfigurationClientWrapper> _clients;
+
+        private IEnumerable<string> _connectionStrings;
+        private IEnumerable<Uri> _endpoints;
+        private TokenCredential _credential;
+        private ConfigurationClientOptions _clientOptions;
+        private Lazy<SrvLookupClient> _srvLookupClient;
+        private DateTimeOffset _lastFallbackClientRefresh = default;
+        private bool _allClientFailed = false;
+
+        private readonly Uri _endpoint;
+        private readonly string _secret;
+        private readonly string _id;
+        private readonly bool _isUsingConnetionString;
+        private readonly bool _replicaDiscoveryEnabled;
+
+        private readonly static TimeSpan _fallbackClientRefreshInterval = TimeSpan.FromMinutes(5);
+
+        public ConfigurationClientManager(AzureAppConfigurationOptions options)
+        {
+            if (options == null)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            if (options.ClientOptions == null)
+            {
+                throw new ArgumentException(nameof(options.ClientOptions));
+            }
+
+            _connectionStrings = options.ConnectionStrings;
+            _endpoints = options.Endpoints;
+            _credential = options.Credential;
+            _clientOptions = options.ClientOptions;
+            _replicaDiscoveryEnabled = options.ReplicaDiscoveryEnabled;
+
+            _srvLookupClient = new Lazy<SrvLookupClient>();
+
+            if (_connectionStrings != null && _connectionStrings.Any())
+            {
+                _isUsingConnetionString = true;
+
+                var connectionString = _connectionStrings.First();
+                _endpoint = new Uri(ConnectionStringUtils.Parse(connectionString, ConnectionStringUtils.EndpointSection));
+                _secret = ConnectionStringUtils.Parse(connectionString, ConnectionStringUtils.SecretSection);
+                _id = ConnectionStringUtils.Parse(connectionString, ConnectionStringUtils.IdSection);
+
+                _clients = _connectionStrings
+                    .Select(connectionString =>

```

Comment:
How does connectionString here not conflict with connectionString defined above?

Reply:
I'm not surprised since it's just declearing a delegate